pkg/azuredisk/controllerserver_v2.go

```diff
@@ -21,29 +21,48 @@ package azuredisk
 import (
 	"context"
 	"fmt"
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2020-12-01/compute"
+	"github.com/container-storage-interface/spec/lib/go/csi"
 	"path"
 	"sort"
 	"strconv"
 	"strings"
 
-	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2020-12-01/compute"
-	"github.com/container-storage-interface/spec/lib/go/csi"
-
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
 	v1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/resource"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
+	clientset "k8s.io/client-go/kubernetes"
 	cloudprovider "k8s.io/cloud-provider"
 	volerr "k8s.io/cloud-provider/volume/errors"
 	"k8s.io/klog/v2"
 
+	azDiskClientSet "sigs.k8s.io/azuredisk-csi-driver/pkg/apis/client/clientset/versioned"
+	"sigs.k8s.io/azuredisk-csi-driver/pkg/provisioner"
 	volumehelper "sigs.k8s.io/azuredisk-csi-driver/pkg/util"
 	"sigs.k8s.io/cloud-provider-azure/pkg/metrics"
 	azure "sigs.k8s.io/cloud-provider-azure/pkg/provider"
+	"sigs.k8s.io/cloud-provider-azure/pkg/retry"
 )
 
+type CloudProvisioner interface {
+	CreateVolume(volumeName string, capacityRange *csi.CapacityRange, volumeCapabilities []*csi.VolumeCapability, parameters map[string]string, secrets map[string]string, volumeContentSource *csi.VolumeContentSource, accessibilityRequirements *csi.TopologyRequirement, ctx context.Context) (*csi.CreateVolumeResponse, error)
```

Comment:
Ideally, the cloud provisioner would have no dependency on CSI types. We can fix this later - add a TODO comment if you defer the change.